protocol/messenger_walletconnect.go

```diff
@@ -30,6 +32,28 @@ func (m *Messenger) AddWalletConnectSession(request *requests.AddWalletConnectSe
 	return m.persistence.InsertWalletConnectSession(session)
 }
 
+func (m *Messenger) NewWalletConnectV2SessionCreatedNotification(session walletconnect.Session) error {
+	now := m.GetCurrentTimeInMillis()
+
+	notification := &ActivityCenterNotification{
+		ID:                         types.FromHex(string(session.Topic) + "_dapp_connected"),
+		Type:                       ActivityCenterNotificationTypeDAppConnected,
+		DAppURL:                    session.Peer.Metadata.URL,
+		DAppName:                   session.Peer.Metadata.Name,
+		WalletProviderSessionTopic: string(session.Topic),
+		Timestamp:                  now,
+		UpdatedAt:                  now,
+	}
+
+	if len(session.Peer.Metadata.Icons) > 0 {
+		notification.DAppIconURL = session.Peer.Metadata.Icons[0]
+	}
+
+	_, err := m.persistence.SaveActivityCenterNotification(notification, true)
```

Comment:
Instead of calling persistence directly, better to use `m.addActivityCenterNotification`.
It will properly set save it, update `MesesngerResponse` and call syncing.
https://github.com/status-im/status-go/blob/8c5a735438238430926a4b768053fccf47ef9594/protocol/messenger_activity_center.go#L264-L265

Search for usages of `addActivityCenterNotification` for examples.